### PR TITLE
catalog-import: add support for multiple git providers

### LINKED ISSUE #3858: Catalog-import should be more generic to support multiple Git providers
Regarding `@backstage/catalog-import`:

> i confirmed the same basic import functionality is working with Bitbucket Server. I do think we need to make the verbiage of the plugin a bit more generic and ideally specific to the integration in use given in the app-config, given someone using Bitbucket wouldn't benefit from all the GitHub verbiage 😀 

> Anything that makes reference to GitHub maybe keep it generic for now and update to something like from "your repository" until we can have it changed to integration.bitbucket = Bitbucket or integration.gitlab = GitLab etc... The Start tracking your component in "Backstage" verbiage should also be updated to "my app" in the app-config, believe that's in the register-component... thoughts? 

_Originally posted by @regicsolutions in https://github.com/backstage/backstage/pull/3782#issuecomment-751716202_

### PR DESCRIPTION
**Pull request by goober**
## Hey, I just made a Pull Request!

This pull request fixes #3858

Screenshot when a single integration is configured and is not Github:
image

Screenshot when multiple integrations are configured and is not GitHub:
image

Screenshot when only Github is enabled:
image

Screenshot when Github and two more integrations are enabled:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by adamdmharvey — APPROVED** [review]
Really nicely done, @goober ! 👏

**Comment by regicsolutions** [comment]
Great work here @goober !!!

**Review by regicsolutions — APPROVED** [review]

File: plugins/catalog-import/src/components/ImportComponentPage.tsx [hunk 1]
```diff
@@ -37,6 +40,28 @@ export type ConfigSpec = {
   config: PartialEntity[];
 };
 
+function manifestGenerationAvailable(configApi: ConfigApi): boolean {
+  return configApi.has('integrations.github');
+}
+
+function repositories(configApi: ConfigApi): string[] {
+  const integrations = configApi.getConfig('integrations');
```
Comment (freben) [inline plugins/catalog-import/src/components/ImportComponentPage.tsx:48]: This should all ideally have used the new `@backstage/integrations` package, but it's not very conveniently exposed on the frontend yet. So OK for now!

**Review by freben — APPROVED** [review]

**Comment by regicsolutions** [comment]
I just installed the nightly release with bitbucket integration in place and am getting this error:

All my locations in my app-config.yaml are set to url only thing that has the word github in it is the annotation set for Jenkins plugin which i recommended we update to be more generic.

```
catalog warn Location 'https://bitbucketserver/projects
/BS/repos/backstage-catalog/browse/software-catalog/
examples/apis/hello-world-api-jenkins.yaml' uses deprecated location type 'github', use 'url' instead.
Use "scripts/migrate-location-types.js" in the Backstage repo to migrate existing locations.

catalog warn Rejecting write of entity API:default/hello-world-api from github:https://bitbucketserver/projects/BS/repos/backstage-catalog/browse
/software-catalog/examples/apis/hello-world-api-jenkins.yaml because entity existed from url:https://bitbucketserver/projects/BS/repos/backstage-catalog/browse/software-catalog/
examples/apis/hello-world-api-jenkins.yaml
```

**Pull request merged**